packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -240,46 +225,92 @@ module AlgebraicCombination = {
     t1: t,
-  ): calculationMethod =>
+  ): calculationStrategy =>
     switch op {
     | #Divide
     | #Power
     | #Logarithm =>
-      MonteCarlo
+      MonteCarloStrat
     | (#Add | #Subtract | #Multiply) as convOp =>
       expectedConvolutionCost(t1) * expectedConvolutionCost(t2) > 10000
-        ? MonteCarlo
-        : Convolution(convOp)
+        ? MonteCarloStrat
+        : ConvolutionStrat(convOp)
     }
 
-  let run = (
+  let tryAnalyticalSimplification = (
+    arithmeticOperation: Operation.algebraicOperation,
+    t1: t,
+    t2: t,
+  ): option<SymbolicDistTypes.analyticalSimplificationResult> => {
+    switch (t1, t2) {
+    | (DistributionTypes.Symbolic(d1), DistributionTypes.Symbolic(d2)) =>
+      Some(SymbolicDist.T.tryAnalyticalSimplification(d1, d2, arithmeticOperation))
+    | _ => None
+    }
+  }
+
+  let runDefault = (
     t1: t,
     ~toPointSetFn: toPointSetFn,
     ~toSampleSetFn: toSampleSetFn,
     ~arithmeticOperation,
     ~t2: t,
   ): result<t, error> => {
     switch tryAnalyticalSimplification(arithmeticOperation, t1, t2) {
-    | Some(Ok(symbolicDist)) => Ok(Symbolic(symbolicDist))
-    | Some(Error(e)) => Error(OperationError(e))
+    | Some(#AnalyticalSolution(symbolicDist)) => Ok(Symbolic(symbolicDist))
+    | Some(#Error(e)) => Error(OperationError(e))
+    | Some(#NoSolution)
     | None =>
       switch getInvalidOperationError(t1, t2, ~toPointSetFn, ~arithmeticOperation) {
       | Some(e) => Error(e)
       | None =>
-        switch chooseConvolutionOrMonteCarlo(arithmeticOperation, t1, t2) {
-        | MonteCarlo => runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
-        | Convolution(convOp) =>
+        switch chooseConvolutionOrMonteCarloDefault(arithmeticOperation, t1, t2) {
+        | MonteCarloStrat => runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
+        | ConvolutionStrat(convOp) =>
+          runConvolution(toPointSetFn, convOp, t1, t2)->E.R2.fmap(r => DistributionTypes.PointSet(
+            r,
+          ))
+        }
+      }
+    }
+  }
+
+  let run = (
+    ~strategy: DistributionTypes.asAlgebraicCombinationStrategy,
+    t1: t,
+    ~toPointSetFn: toPointSetFn,
+    ~toSampleSetFn: toSampleSetFn,
+    ~arithmeticOperation: Operation.algebraicOperation,
+    ~t2: t,
+  ): result<t, error> => {
+    switch strategy {
+    | AsDefault => runDefault(t1, ~toPointSetFn, ~toSampleSetFn, ~arithmeticOperation, ~t2)
+    | AsSymbolic =>
+      switch tryAnalyticalSimplification(arithmeticOperation, t1, t2) {
+      | Some(#AnalyticalSolution(symbolicDist)) => Ok(Symbolic(symbolicDist))
+      | Some(#NoSolution) => Error(RequestedStrategyInvalidError(`No analytical solution`))
+      | None => Error(RequestedStrategyInvalidError("Inputs were not even symbolic"))
+      | Some(#Error(err)) => Error(OperationError(err))
+      }
+    | AsConvolution => {
+        let errString = opString => `Can't convolve on ${opString}`
+        switch arithmeticOperation {
```

Comment:
instead of nesting the switch statements, its often better to have one level.

```
switch (strategy, arithmeticOperation) {
| (AsDefault, _) =>
```

Reply:
I'm not sure I like pattern matching where most of the cases are `_`, also each branch switches into a different thing, so it's not a clear improvement.